bindings/experimental/distrdf/python/DistRDF/__init__.py

```diff
@@ -63,6 +63,69 @@ def create_logger(level="WARNING", log_path="./DistRDF.log"):
     return logger
 
 
+def RunGraphs(proxies, concurrentruns=4):
+    """
+    Trigger the execution of multiple RDataFrame computation graphs on a certain
+    distributed backend. If the backend doesn't support multiple job
+    submissions concurrently, the distributed computation graphs will be
+    executed sequentially.
+
+    Args:
+        proxies(list): List of action proxies that should be triggered. Only
+            actions belonging to different RDataFrame graphs will be
+            triggered to avoid useless calls.
+
+        concurrent_runs(int, optional): Number of graphs that will be
+            executed concurrently in a distributed backend. Defaults to 4.
+
+    Exceptions:
+
+        TypeError: All actions need to belong to the same type of
+        distributed backend. The function raises an error if this condition is
+        not met.
+
+    Example:
+
+        @code{.py}
+        import ROOT
+        RDataFrame = ROOT.RDF.Experimental.Distributed.Spark.RDataFrame
+        RunGraphs = ROOT.RDF.Experimental.Distributed.RunGraphs
+
+        # Create 3 different dataframes and book an histogram on each one
+        histoproxies = [
+            RDataFrame(100)
+                .Define("x", "rdfentry_")
+                .Histo1D(("name", "title", 10, 0, 100), "x")
+            for _ in range(4)
+        ]
+
+        # Execute the 3 computation graphs
+        RunGraphs(histoproxies)
+        # Retrieve all the histograms in one go
+        histos = [histoproxy.GetValue() for histoproxy in histoproxies]
+        @endcode
+
+
+    """
+
+    # Get proxies belonging to distinct computation graphs
+    uniqueproxies = list({proxy.proxied_node.get_head(): proxy for proxy in proxies}.values())
+
+    expectedbackend = type(uniqueproxies[0].proxied_node.get_head().backend)
+    for proxy in uniqueproxies:
+        if type(proxy.proxied_node.get_head().backend) is not expectedbackend:
+            raise TypeError(("Actions provided to RunGraphs need to be executable"
```

Comment:
I think this is fair, at least in first approximation. However I like the idea of running different RDF graphs in different resources with different backends: related to my comment above about the possible lack of parallelization with Spark, if we allowed to run different graphs with different backends (provided we have enough resources, of course) this would open the door to parallelize on multiple infrastructures. Even if it's nice that the infrastructure itself manages elasticity (and not the user), this might be useful for power users.

Reply:
This can be done but there must be a way to submit the different concurrent submissions to the different backends in parallel. So either another threadpool or some asynchronous submission in this backend-agnostic function

Reply:
Yes you would need to introduce some extra asynchrony here.